deps/verifier/src/cca/remote.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2023 Arm Ltd.
+// Copyright (c) 2025 Linaro Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use super::*;
+use base64::Engine;
+use config::Config;
+use core::result::Result::Ok;
+use ear::Ear;
+use log::{debug, error};
+use std::str;
+
+const MEDIA_TYPE: &str = r#"application/eat-collection; profile="http://arm.com/CCA-SSD/1.0.0""#;
+
+fn evidence_builder(
+    nonce: &[u8],
+    accept: &[String],
+    token: Vec<u8>,
+) -> Result<(Vec<u8>, String), veraison_apiclient::Error> {
+    debug!("server challenge: {:?}", nonce);
+    debug!("acceptable media types: {:#?}", accept);
+    // TODO: Get the CCA media type from the slice of `accept`.
+    Ok((token, MEDIA_TYPE.to_string()))
+}
+
+pub async fn verify(
+    config: Config,
+    token: &[u8],
+    expected_report_data: &Vec<u8>,
```

Comment:
I'm also curious why local and remote verifier don't have same algorithm. shouldn't the only difference is the method to get the certificates?

Reply:
The appraisal algorithm is (in theory) the same, except it’s hidden in two different black boxes:
1. For local, the rust-ccatoken evidence appraisal function,
2. For remote, the Veraison CCA appraisal plugin.

Given the same evidence payload, the resulting trustworthiness vector should be identical.

A slight difference is the freshness check, which Veraison handles as part of the appraisal, whereas in “local” mode it needs to be done manually.

Note: CCA platform verification does not use certificates (in the sense of X.509 certs).  Platform evidence is verified using a raw public key endorsed alongside the platform instance identifier.